File: backend/judge_files/python3/yeetcode/ListNode.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def deserialize_list(arr):
    head = ListNode()
    cur = head
    for val in arr:
        cur.next = ListNode(val)
        cur = cur.next
    return head.next

def serialize_list(head):
    arr = []
    while head:
        arr.append(head.val)
        head = head.next
    return arr

File: backend/judge_files/javascript/yeetcode/listnode.js
class ListNode {
    constructor(val = 0, next = null) {
        this.val = val;
        this.next = next;
    }
}

function deserializeList(arr) {
    let head = new ListNode();
    let cur = head;
    for (let i = 0; i < arr.length; i++) {
        cur.next = new ListNode(arr[i]);
        cur = cur.next;
    }
    return head.next;
}

function serializeList(head) {
    let arr = [];
    while (head) {
        arr.push(head.val);
        head = head.next;
    }
    return arr;
}

module.exports = { ListNode, deserializeList, serializeList };